pkg/controller/security/manager_test.go

```diff
@@ -123,12 +123,13 @@ func runTestCertRotate(t *testing.T) {
 	time.Sleep(2 * time.Second)
 	for {
 		cert2 := secretManager.GetCert(identity1)
+		secretManager.certsCache.mu.RLock()
```

Comment:
This code looks strange, `cert2` is not protected between `secretManager.GetCert(identity1)` and this line.

Reply:
Any good suggestion?